### ISSUE TITLE
sqlite3.OperationalError: no such table: xonsh_history

### ISSUE THREAD

**Issue by astrogewgaw** [xonsh/xonsh#4593]
<div align="justify">

The SQLite history backend keeps crashing. I am documenting a particular instance below, but this particular error has plagued me for quite some time, and across multiple Linux distros: Ubuntu, Manjaro and Arch. I am using the `kitty` terminal. I have also shared my `.xonshrc`, if it helps anyone diagnose the problem.

**NOTE**: I have also noticed that `xonsh` starts creating a lot of files named `None` in the current working directory, *irrespective of the backend I use*. Whenever that file is deleted, the same error pops up again. I believe that `xonsh` is having some problems with creating the SQLite/JSON history tables/files in the first place.

</div>

## xonshrc

<details>

```python
$DOTGLOB = True
$AUTO_CD = True
$EDITOR = "nvim"
$MULTILINE_PROMPT=" "
$MANPAGER = "less -X"
$HISTCONTROL="ignoredups"
$XONSH_HISTORY_BACKEND = "sqlite"
$LESS = "--ignore-case --quit-if-one-screen --quit-on-intr FRXQ"

$HDD = "/run/media/$USER/codex"

aliases["-"] = "cd -"
aliases[".."] = "cd .."
aliases["...."] = "cd ../.."

aliases["ll"] = "exa -l@h"
aliases["ls"] = "exa -l@h"
aliases["la"] = "exa -l@ha"
aliases["lt"] = "exa -l@hT"

aliases["vim"] = "nvim"
aliases["hdd"] = ["cd", $HDD]
aliases["school"] = ["cd", f"{$HDD}/school"]
aliases["ssh"] = ["kitty", "+kitten", "ssh"]
aliases["wally"] = ["wal", "-i", f"{$HOME}/wallpapers"]
aliases["cp"] = ["rsync", "--progress", "--recursive", "--archive"]
aliases["md"] = lambda args: execx(f"mkdir -p {repr(args[0])} && cd {repr(args[0])}")


plugins = [
	"sh",
	"mpl",
	"kitty",
	"autoxsh",
	"onepath",
	"pipeliner",
	"fzf-widgets",
	"argcomplete",
	"prompt_starship",
	"apt_tabcomplete",
	"readable-traceback",
]
if plugins:
	xontrib load @(plugins)
	
$fzf_ssh_binding = "c-s"		# Ctrl+S
$fzf_dir_binding = "c-g"		# Ctrl+G        
$fzf_file_binding = "c-t"		# Ctrl+T
$fzf_history_binding = "c-r"		# Ctrl+R

for path in [
	p"/bin",
	p"/sbin",
	p"/usr/sbin",
	p"/usr/bin",
	p"/snap/bin",
	p"/usr/games",
	p"/usr/local/bin",
	p"/usr/local/sbin",
	p"/usr/local/games",
	p"/usr/local/go/bin",
	p"/home/$USER/go/bin",
	p"/home/$USER/conda/bin",
	p"~/.local/kitty.app/bin",
	p"/home/$USER/.local/bin",
	p"/home/$USER/.cargo/bin",
	p"/home/$USER/.pyenv/bin",
	p"/home/linuxbrew/.linuxbrew/bin",
	p"/home/linuxbrew/.linuxbrew/sbin",
]:
	if path.exists():
		$PATH.append(path)



# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
import sys as _sys
from types import ModuleType as _ModuleType
_mod = _ModuleType("xontrib.conda",
                   "Autogenerated from $(/home/neo/conda/bin/conda shell.xonsh hook)")
__xonsh__.execer.exec($("/home/neo/conda/bin/conda" "shell.xonsh" "hook"),
                      glbs=_mod.__dict__,
                      filename="$(/home/neo/conda/bin/conda shell.xonsh hook)")
_sys.modules["xontrib.conda"] = _mod
del _sys, _mod, _ModuleType
# <<< conda initialize <<<
```

</details>

## xonfig

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.11.0               |
| Git SHA          | 337cf25a             |
| Commit Date      | Nov 17 15:37:41 2021 |
| Python           | 3.9.5                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.23               |
| shell type       | prompt_toolkit       |
| history backend  | sqlite               |
| pygments         | 2.10.0               |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on wsl           | False                |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
| xontrib 1        | apt_tabcomplete      |
| xontrib 2        | argcomplete          |
| xontrib 3        | autoxsh              |
| xontrib 4        | conda                |
| xontrib 5        | fzf-widgets          |
| xontrib 6        | kitty                |
| xontrib 7        | mpl                  |
| xontrib 8        | onepath              |
| xontrib 9        | pipeliner            |
| xontrib 10       | prompt_starship      |
| xontrib 11       | readable-traceback   |
| xontrib 12       | sh                   |
| RC file 1        | /home/neo/.xonshrc   |
+------------------+----------------------+
```

</details>

## Expected Behavior

The command below should make a directory and `cd` into it:

```bash
md <SOME DIRECTORY>
```

The above command is an alias for:

```python
lambda args: execx(f"mkdir -p {repr(args[0])} && cd {repr(args)[0])}")
```

## Current Behavior

It crashes with an error.

### Traceback

This is the traceback I get with `$XONSH_DEBUG=1`.

<details>

```bash
<xonsh-code>:1:3 - md ~/.config/xonsh/rc.d/
<xonsh-code>:1:3 + ![md ~/.config/xonsh/rc.d/]
<xonsh-code>:1:9:42 - mkdir -p '/home/neo/.config/xonsh/rc.d/' && cd '/home/neo/.config/xonsh/rc.d/'
<xonsh-code>:1:9:42 + ![mkdir -p '/home/neo/.config/xonsh/rc.d/'] && cd '/home/neo/.config/xonsh/rc.d/'
<xonsh-code>:1:50 - ![mkdir -p '/home/neo/.config/xonsh/rc.d/'] && cd '/home/neo/.config/xonsh/rc.d/'
<xonsh-code>:1:50 + ![mkdir -p '/home/neo/.config/xonsh/rc.d/'] && ![cd '/home/neo/.config/xonsh/rc.d/']
SQLite History Backend Error: no such table: xonsh_history
prompt: error: on field 'starship_left'
Traceback (most recent call last):
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/prompt/__amalgam__.py", line 924, in _no_cache_field_value
    value = field_value() if callable(field_value) else field_value
  File "/home/neo/conda/lib/python3.9/site-packages/xontrib/prompt_starship.py", line 25, in <lambda>
    __xonsh__.env['PROMPT_FIELDS']['starship_left']	= lambda: _starship_prompt(_left_cfg)
  File "/home/neo/conda/lib/python3.9/site-packages/xontrib/prompt_starship.py", line 13, in _starship_prompt
    '--status', str(int( __xonsh__.history[-1].rtn)) if len(__xonsh__.history) > 0 else '0',
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/history/__amalgam__.py", line 134, in __len__
    return len(list(self.items()))
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/history/__amalgam__.py", line 1183, in items
    yield from self.all_items(newest_first, session_id=str(self.sessionid))
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/history/__amalgam__.py", line 1176, in all_items
    for inp, ts, rtn, freq, cwd in xh_sqlite_items(
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/history/__amalgam__.py", line 1047, in xh_sqlite_items
    return _xh_sqlite_get_records(c, sessionid=sessionid, newest_first=newest_first)
  File "/home/neo/conda/lib/python3.9/site-packages/xonsh/history/__amalgam__.py", line 1009, in _xh_sqlite_get_records
    cursor.execute(sql, tuple(params))
sqlite3.OperationalError: no such table: xonsh_history
```

</details>

## Steps to Reproduce

<div align="justify">

I am not exactly sure what triggers this. The command given above triggered it for me, but it may not do so for others. If someone can reproduce this error, do let me know by dropping a comment below 😁 !

</div>

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
Hi! I can't repeat this with your xonshrc. 

Please run:
```python
history info

echo Im @$(whoami)
f = $XONSH_HISTORY_FILE
while str(f) != '/':
    stat -c '%a %U %G - %n' @(f)
    f = f.parent
```
And copy here the output.

**Comment by astrogewgaw** [comment]
<div align="justify">

Thanks for responding so quickly, @anki-code! I have moved away from `xonsh` for the mean time, but I will try the steps you have outlined above and let you know how it goes. As for the specific alias I mentioned, I don't think it is the trigger. `xonsh` is having some problems in making the SQLite database and table where it should. It somehow keeps making a file called `None` in each working directory I `cd` into, and then malfunctions (obviously) if I delete it. I will experiment with a cleaner install of xonsh (without my `.xonshrc`) and let you know if I run into the same error.

</div>

**Comment by anki-code** [comment]
I'm using Arch as well and I have no any problems with xonsh ever.
I think you have wrong permissions for the xonsh directory in your home.

**Comment by dev2718** [comment]
Based on 
> NOTE: I have also noticed that xonsh starts creating a lot of files named None in the current working directory, irrespective of the backend I use. Whenever that file is deleted, the same error pops up again. I believe that xonsh is having some problems with creating the SQLite/JSON history tables/files in the first place.

It is highly likely that this was fixed with #4649, which got rid of these "None" history files in the current working directory. (See also #4600).
